prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -297,24 +298,37 @@ def prompt_processor() -> Any:
                 doc_id=doc_id,
                 usage_kwargs=usage_kwargs,
             )
-            if context is None:
-                # TODO: Obtain user set name for vector DB
-                msg = "Couldn't fetch context from vector DB"
-                app.logger.error(
-                    f"{msg} {output[PSKeys.VECTOR_DB]} for doc_id {doc_id}"
+            if not context:
+                # UN-1288 For Pinecone, we are seeing an inconsistent case where
+                # query with doc_id fails even though indexing just happened.
+                # This causes the following retrieve to return no text.
+                # To rule out any lag on the Pinecone vector DB write,
+                # the following sleep is added
+                time.sleep(2)
+                context: Optional[str] = tool_index.get_text_from_index(
```

Comment:
@gaya3-zipstack This sleep would be triggered for all adapters right?
I would suggest if there is a validation to use this workaround just for Pinecone alone. I understand, it is a 2sec sleep time and retry, but will this be into all other adapter flows which does not return a context?

Reply:
@harini-venkataraman Adding it for pinecone alone will fit right only if done in adapters. To do it right, we will ideally need to pass a retry param to the bottom-most function and sleep there. However as we know, this will need more testing and also sdk/adapter version roll. So as a quick fix,  this is the top-most place where we are seeing the issue and hence adding it for now like a safety net for the demo to consumer.

Reply:
@gaya3-zipstack Get that. I hope we have some mechanism to track this.

Reply:
@gaya3-zipstack if we have context on which type of adapter is being used for this retrieval - we can simply add this sleep for Pinecone as @harini-venkataraman mentioned right?

Reply:
@chandrasekharan-zipstack I strongly agree. I am hoping we remove this piece of code in the coming days. The issue is we are unable to reproduce the problem with a set pattern. Sometimes it works and sometimes it does not. This is an experimental fix. If it works fine with this, then we should move this to adapters with a retry logic built with a vector db query function actually. Due to lack of time to give the clean fix, I have added it here.